docs/architecture/adr-107-betaize-proto-versions.md

```diff
@@ -0,0 +1,85 @@
+# ADR 107: Rename proto versions preceding 1.0 to pre-v1 betas
+
+## Changelog
+
+- 2023-07-11: Initial draft
+
+## Status
+
+Proposed (CHANGE BEFORE MERGING)
+
+## Context
+
+The drive to introduce [protobuf versioning][cometbft#95] resulted in
+introducing versioned packages for protobuf definitions corresponding to
+each of the major CometBFT releases starting from 0.34. By the upcoming 0.39
+release, some of the packages will get up to `v4`, as the development churn
+and the intent to perform code style grooming have resulted in
+backward-incompatible changes every time. All this storied history
+has not yet been released with semver commitments on [buf schema registry][bsr]
+or even merged into the main branch at the time of this writing.
+
+Efforts to conform to the [buf style guide][buf-style]
+(started with [#736][cometbft#736]) have been confined to latter versions
+of the proto packages in order to preserve source code compatibility
+and ease migration for developers. The earlier packages, therefore, do not
+constitute exemplary protobuf material and may even be rejected by a schema
+repository linter.
+
+## Alternative Approaches
+
+We can do nothing and go ahead with the current versioning rework as per
+[ADR-103], which does solve the main problem of managing backward-incompatible
```

Comment:
Nit: If this is intended to hyperlink to the reference introduced on L81, it doesn't work. Same for the link on L53.